share/picongpu/examples/IncidentField/include/picongpu/param/incidentField.param

```diff
@@ -33,42 +33,64 @@ namespace picongpu
+                static constexpr float_64 TIME_PERIOD_SI = 0.038e-9; // 0.038 ns
+                static constexpr float_64 WAVE_LENGTH_SI = TIME_PERIOD_SI * SI::SPEED_OF_LIGHT_SI;
+                static constexpr float_64 UNITCONV_A0_to_Amplitude_SI = -1.0 / WAVE_LENGTH_SI * SI::ELECTRON_MASS_SI
+                    * SI::SPEED_OF_LIGHT_SI * SI::SPEED_OF_LIGHT_SI / SI::ELECTRON_CHARGE_SI;
                 static constexpr float_64 A0 = 8.0;
                 static constexpr float_64 AMPLITUDE_SI = A0 * UNITCONV_A0_to_Amplitude_SI;
-                static constexpr uint32_t DURATION_STEPS = 300;
+                static constexpr float_64 DURATION_SI = 0.6e-9; // 0.6 ns
+
                 /// These need to be changed together with grid size
-                static constexpr uint32_t CENTER_X_CELLS = 80;
-                static constexpr uint32_t WIDTH_X_CELLS = 40;
-                static constexpr uint32_t CENTER_Y_CELLS = 40;
-                static constexpr uint32_t WIDTH_Y_CELLS = 20;
+                static constexpr float_64 CENTER_X_SI = 80.0e-3; // 80 mm
+                static constexpr float_64 WIDTH_X_SI = 40.0e-3; // 40 mm
+                static constexpr float_64 CENTER_Y_SI = 40.0e-3; // 40 mm
+                static constexpr float_64 WIDTH_Y_SI = 20.0e-3; // 20 mm
 
-                HDINLINE float_X getTimeComponentSI(const float_X currentStep) const
-                {
-                    auto const durationStepsMiddle = DURATION_STEPS / 2;
-                    auto const normalizedDuration = (currentStep - durationStepsMiddle) / durationStepsMiddle;
-                    auto const timeSI = currentStep * SI::DELTA_T_SI;
-                    const float_X sinArg = precisionCast<float_X>(timeSI / TIME_PERIOD_SI * 2.0 * PI);
-                    return AMPLITUDE_SI * math::exp(-normalizedDuration * normalizedDuration) * math::sin(sinArg);
-                }
+                // Convert to grid units
+                static constexpr uint32_t DURATION_STEPS = DURATION_SI / SI::DELTA_T_SI;
+                static constexpr uint32_t CENTER_X_CELLS = CENTER_X_SI / SI::CELL_WIDTH_SI;
+                static constexpr uint32_t WIDTH_X_CELLS = WIDTH_X_SI / SI::CELL_WIDTH_SI;
+                static constexpr uint32_t CENTER_Y_CELLS = CENTER_Y_SI / SI::CELL_HEIGHT_SI;
+                static constexpr uint32_t WIDTH_Y_CELLS = WIDTH_Y_SI / SI::CELL_HEIGHT_SI;
 
-                HDINLINE float_X getTransversalXComponentSI(const floatD_X& totalCellIdx) const
+                // Get value to calculate Ey / Bz for the XMin source
+                HDINLINE float_X getXMinSourceValueSI(const floatD_X& totalCellIdx, const float_X currentStep) const
                 {
-                    auto const normalizedX = (totalCellIdx[0] - CENTER_X_CELLS) / WIDTH_X_CELLS;
-                    return math::exp(-normalizedX * normalizedX);
+                    auto const initializationDurationSI = 16.0 * DURATION_SI;

```

Comment:
As you define the variable `initializationDurationSI` the second time here, you could as well just define it as a member of the class. (Since the number 16 is used here, I think a central definition is more consistent)